src/Microsoft.AspNetCore.Hosting/Internal/HostingEnvironmentExtensions.cs

```diff
@@ -3,26 +3,42 @@
 
 using System;
 using System.IO;
-using Microsoft.AspNetCore.Hosting.Internal;
-using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.PlatformAbstractions;
 
 namespace Microsoft.AspNetCore.Hosting.Internal
 {
     public static class HostingEnvironmentExtensions
     {
-        public static void Initialize(this IHostingEnvironment hostingEnvironment, string applicationBasePath, WebHostOptions options, IConfiguration configuration)
+        public static void Initialize(this IHostingEnvironment hostingEnvironment, IApplicationEnvironment applicationEnvironment, WebHostOptions options)
         {
             if (options == null)
             {
                 throw new ArgumentNullException(nameof(options));
             }
 
+            hostingEnvironment.ApplicationName = applicationEnvironment.ApplicationName;
+
+            hostingEnvironment.ContentRootPath = applicationEnvironment.ApplicationBasePath;
+            if (!string.IsNullOrEmpty(hostingEnvironment.ContentRootPath))
```

Comment:
Some of these checks are redundant. We always set ContentRootPath in IApplicationEnvironment to a rooted path so null/empty check and getting full path is not required. Might remove.